code/modules/clothing/glasses/glasses.dm

```diff
@@ -272,6 +275,16 @@ var/list/science_goggles_wearers = list()
 	eyeprot = 1
 	species_fit = list(VOX_SHAPED, GREY_SHAPED, INSECT_SHAPED)
 
+/obj/item/clothing/glasses/sunglasses/equipped(mob/M, slot)
+	if (M.self_vision)
+		M.self_vision.target_alpha = SUNGLASSES_TARGET_ALPHA // You see almost nothing with those on!
```

Comment:
Ditto, the `self_vision` layer is painted on the backdrop. But that backdrop doesn't work if the rest of the lighting follows different rules. It's not visible.